pkg/provider/acceptance_source_webhook_test.go

```diff
@@ -0,0 +1,88 @@
+package provider
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/materialize"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/jmoiron/sqlx"
+)
+
+func TestAccSourceWebhook_basic(t *testing.T) {
+	sourceName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSourceWebhookResource(sourceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSourceWebhookExists("materialize_source_webhook.test"),
+					resource.TestCheckResourceAttr("materialize_source_webhook.test", "name", sourceName),
+					resource.TestCheckResourceAttr("materialize_source_webhook.test", "cluster_name", sourceName),
+					resource.TestCheckResourceAttr("materialize_source_webhook.test", "body_format", "json"),
+					resource.TestCheckResourceAttr("materialize_source_webhook.test", "include_headers", "false"),
+				),
+			},
+			{
+				ResourceName:      "materialize_source_webhook.test",
+				ImportState:       true,
+				ImportStateVerify: false,
+			},
+		},
+	})
+}
+
+func testAccSourceWebhookResource(sourceName string) string {
+	return fmt.Sprintf(`
+resource "materialize_cluster" "example_cluster" {
+	name = "%[1]s"
+	size = "1"
+	replication_factor = 1
+}
+
+resource "materialize_source_webhook" "test" {
+	name = "%[1]s"
+	cluster_name = materialize_cluster.example_cluster.name
+	body_format = "json"
+	include_headers = false
+}
+`, sourceName)
+}
+
+func testAccCheckSourceWebhookExists(name string) resource.TestCheckFunc {
```

Comment:
Realizing we could just use the same function for all source acceptance tests rather than defining a custom one for each source type. At least for any source that just uses `ScanSource`.